entry/src/main/java/com/thoughtbot/expandablerecyclerview/multicheck/MultiCheckHelper.java

```diff
@@ -0,0 +1,295 @@
+                                ShapeElement.OVAL, Image.ScaleMode.CENTER, ResourceTable.Color_white);
+                    }
+                } else {
+                    // Add child items to list
+                    holder.makeInvisibleImage(ResourceTable.Id_ArrowIcon);
+                    holder.setText(ResourceTable.Id_tvGroupTitle, text.getChildItem(), Color.BLACK,
+                            ResUtil.getIntDimen(context, ResourceTable.Float_child_text_size));
+                    boolean flag = false;
+                    for (int i = 0; i < mSelectedChild.size(); i++) {
+                        if (mSelectedChild.get(i).isEquals(text)) {
+                            flag = true;
+                            break;
+                        }
+                    }
+                    if (flag) {
+                        holder.setChecked(ResourceTable.Id_checkbtn);
+                    } else {
+                        holder.setUnChecked(ResourceTable.Id_checkbtn);
+                    }
+                }
+            }
+        };
+        mGroupContainer.setItemProvider(expandableListAdapter);
+
+        expandableListAdapter.setOnItemClickListener((component, position) -> {
+            ParentChild value = mGroupNameItem.get(position);
+            String clickedItem = value.getChildItem();
+            if (!mTempChildNameItem.contains(clickedItem)) {
+                if (mTempGroupNameItem.contains(clickedItem)) {
+                    int actualItemPosition = mFinalGroupNameItem.indexOf(clickedItem);
+                    removeChildItems(actualItemPosition, position);
+                    mTempGroupNameItem.remove(clickedItem);
+                } else {
+                    int actualItemPosition = mFinalGroupNameItem.indexOf(clickedItem);
+                    addChildItems(actualItemPosition, clickedItem, position);
+                    mTempGroupNameItem.add(clickedItem);
+                }
+                expandableListAdapter.setData(mGroupNameItem);
+
+            } else {
+                int index = -1;
+                for (int i = 0; i < mSelectedChild.size(); i++) {
+                    if (mSelectedChild.get(i).isEquals(value)) {
+                        index = i;
+                        break;
+                    }
+                }
+                if (index != -1) {
+                    mSelectedChild.remove(index);
+                } else {
+                    mSelectedChild.add(value);
+                }
+                expandableListAdapter.setData(mGroupNameItem);
+            }
+        });
+
+        //To clear the all the selected items
+        clearbtn.setClickedListener(component -> {
+            mSelectedChild.clear();
+            expandableListAdapter.setData(mGroupNameItem);
+        });
+
+        //To automatically check boston child item under Parent Rock
+        bostonbtn.setClickedListener(component -> {
+            boolean flag = true;
```

Comment:
use meaningful naming conventions for better readability. 
Instead of flag, name it as follows:

boolean isChecked = false;
ParentChild value = new ParentChild(valueRock, valueBoston);
            for (int i = 0; i < mSelectedChild.size(); i++) {
                if (mSelectedChild.get(i).isEquals(value)) {
                    isChecked = true;
                    break;
                }
            }
            if (!isChecked ) {
                mSelectedChild.add(new ParentChild(valueRock, valueBoston));